src/app/redux/reducer.js

```diff
@@ -0,0 +1,30 @@
+import * as types from './constants';
+
+const initialState = {
+    cities: JSON.parse(localStorage.getItem('latestSearches')) || []
+}
+
+export default function (state = initialState, action) {    
+    switch (action.type) {
+        case types.ADD_CITY: {
+            addCity(state, action);
+        }
+    }
+
+    return state;
+}
+
+const addCity = (state, action) => {
+    action.city = action.city.trim().toLowerCase();
+    if (state.cities.includes(action.city)) {
+        const index = state.cities.indexOf(action.city);
```

Comment:
Why do you check twice if exis?? I think just removing the first if is also okay 👍